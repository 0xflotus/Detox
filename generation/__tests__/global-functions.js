const globals = require("../earl-grey/global-functions");

describe("globals", () => {
  describe("sanitize_greyDirection", () => {
    it('should return numbers for strings', () => {
      expect(globals.sanitize_greyDirection('left')).toBe(1);
      expect(globals.sanitize_greyDirection("right")).toBe(2);
      expect(globals.sanitize_greyDirection("up")).toBe(3);
      expect(globals.sanitize_greyDirection("down")).toBe(4);
    });

    it('should fail with unknown value', () => {
      expect(() => {
        globals.sanitize_greyDirection("kittens");
      }).toThrowErrorMatchingSnapshot();
    });
  });
});