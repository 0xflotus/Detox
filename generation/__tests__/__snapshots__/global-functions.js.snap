// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globals sanitize_greyDirection should fail with unknown value 1`] = `"GREYAction.GREYDirection must be a 'left'/'right'/'up'/'down', got kittens"`;
